core/src/main/java/com/kin/ecosystem/core/data/blockchain/BlockchainSourceImpl.java

```diff
@@ -372,17 +398,19 @@ public KeyStoreProvider getKeyStoreProvider() {
 
 	@Override
 	public void updateActiveAccount(int accountIndex) throws BlockchainException {
-		local.setAccountIndex(accountIndex);
-		createKinAccountIfNeeded();
-		
-		synchronized (balanceObserversLock) {
-			removeRegistration(balanceRegistration);
-			if (balanceObserversCount > 0) {
-				startBalanceListener();
+		if (activeAccountIndex != accountIndex && accountIndex != -1) {
+			local.setAccountIndex(accountIndex);
```

Comment:
also, set activeAccountIndex otherwise createKinAccountIfNeeded will try to create previous index

Reply:
create private setAccountIndex that set both in memory and local